lib/cluster/node.js

```diff
@@ -0,0 +1,979 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2020 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const assert = require('assert');
+
+const getIP = require('ip');
+const Bluebird = require('bluebird');
+const EventEmitter = require('eventemitter3');
+const _ = require('lodash');
+
+const debug = require('../util/debug')('kuzzle:cluster:sync');
+const { Mutex } = require('../util/mutex');
+const { ClusterIdCardHandler } = require('./idCardHandler');
+const ClusterPublisher = require('./publisher');
+const ClusterSubscriber = require('./subscriber');
+const ClusterState = require('./state');
+const ClusterCommand = require('./command');
+const kuzzleStateEnum = require('../kuzzle/kuzzleStateEnum');
+
+/**
+ * @typedef {nodeActivityEnum}
+ */
+const nodeActivityEnum = Object.freeze({
+  ADDED: 1,
+  EVICTED: 2,
+});
+
+// Handles the node logic: discovery, eviction, heartbeat, ...
+// Dependencies: core:cache module must be started
+class ClusterNode {
+  constructor () {
+    this.config = global.kuzzle.config.cluster;
+    this.heartbeatDelay = this.config.heartbeat;
+
+    checkConfig(this.config);
+
+    this.ip = getIP.address('public', this.config.ipv6 ? 'ipv6' : 'ipv4');
+
+    this.nodeId = null;
```

Comment:
Shouldn't this property named `idCard` instead ? It was confusing to me because I thought it was the node string ID

Reply:
It IS the node string ID :thinking: 
What makes you think otherwise?